src/lib.rs

```diff
@@ -616,8 +662,24 @@ To override the toolchain mounted in the image, set `target.{}.image.toolchain =
             } else if !args.all.iter().any(|a| a.starts_with("--target")) {
                 let mut args_with_target = args.all.clone();
                 args_with_target.push("--target".to_owned());
-                args_with_target.push(target.triple().to_owned());
+                args_with_target.push(add_libc(target.triple()));
                 args_with_target
+            } else if zig_version.is_some() {
+                let mut filtered_args = Vec::new();
+                let mut args_iter = args.all.clone().into_iter();
+                while let Some(arg) = args_iter.next() {
+                    if arg == "--target" {
+                        filtered_args.push("--target".to_owned());
+                        if let Some(triple) = args_iter.next() {
+                            filtered_args.push(add_libc(&triple));
+                        }
+                    } else if let Some(stripped) = arg.strip_prefix("--target=") {
+                        filtered_args.push(format!("--target={}", add_libc(stripped)));
+                    } else {
+                        filtered_args.push(arg);
+                    }
+                }
+                filtered_args
```

Comment:
do we need to add the zig specific `*.ver` here?

Reply:
Yes, because otherwise we have to add it much, much later, and this doesn't change the `target`, or any other variables except those in the filtered args.

We could technically do it much later, but then we'd have to modify it for both remote and local, and we'd have to re-handle the logic of filtering the args. If we do it earlier, during CLI parsing, then we get the wrong target triple.

Reply:
I guess technically we could do it when adding it to the CLI arguments, but that feels iffy because it means modifying the logic, since we'd need to have already parsed `Cross.toml` which moves that significantly further up, and we'd have to then pass the Zig glibc version to the target callback. Which, IMO, feels odd.